x-pack/filebeat/input/httpjson/value_tpl.go

```diff
@@ -329,10 +331,58 @@ func hmacStringBase64(hmacType string, hmacKey string, values ...string) string
 	}
 	bytes := hmacString(hmacType, []byte(hmacKey), data)
 
-	// Get result and encode as hexadecimal string
+	// Get result and encode as base64 string
 	return base64.StdEncoding.EncodeToString(bytes)
 }
 
+func hashStringHex(typ string, values ...string) string {
+	// Get result and encode as hexadecimal string
+	h := hashStrings(typ, values)
+	if h == nil {
+		return ""
+	}
+	return hex.EncodeToString(h)
+}
+
+func hashStringBase64(typ string, values ...string) string {
+	// Get result and encode as base64 string
+	h := hashStrings(typ, values)
+	if h == nil {
+		return ""
+	}
+	return base64.StdEncoding.EncodeToString(h)
+}
+
+func hashStrings(typ string, data []string) []byte {
+	if len(data) == 0 {
+		return nil
+	}
+	var n int
+	for _, d := range data {
+		n += len(d)
+	}
+	if n == 0 {
+		return nil
```

Comment:
The current behaviour treats "" specially (I think this is wrong, but I'm keeping that behaviour).

Reply:
I think we should remove the special handling for `hash`.

I just looked at the hmac implementation and don't know why the special handling was added there. @marc-gr would there be a reason to handle empty values with special behavior?

Reply:
The only reason I can remember of is since this will be used from templates an empty value would come as <no value>, but now that I try to think about anything else I am leaning more towards a bad outcome of trying to be too defensive. I'd rather change the way we deal with it in the other functions.